src/System.Runtime/tests/System/GCTests.netcoreapp.cs

```diff
@@ -23,5 +25,88 @@ public static void GetAllocatedBytesForCurrentThread(int size)
             Assert.True((end - start) > size, $"Allocated too little: start: {start} end: {end} size: {size}");
             Assert.True((end - start) < 5 * size, $"Allocated too much: start: {start} end: {end} size: {size}");
         }
+
+        [Fact]
+        public static void GetGCMemoryInfo()
+        {
+            GCMemoryInfo memoryInfo = GC.GetGCMemoryInfo();
+
+            Assert.True(memoryInfo.HighMemoryLoadThresholdBytes > 0);
+            Assert.True(memoryInfo.MemoryLoadBytes > 0);
+            Assert.True(memoryInfo.TotalAvailableMemoryBytes > 0);
+            Assert.True(memoryInfo.HeapSizeBytes > 0);
+            Assert.True(memoryInfo.FragmentedBytes >= 0);
+        }
+
+        [Fact]
+        public static void GetGCMemoryInfoFragmentation()
+        {
+            RemoteExecutor.Invoke(() =>
+            {
+                GCHandle[] gch = new GCHandle[100];
+
+                try
+                {
+                    // Allows to update the value returned by GC.GetGCMemoryInfo
+                    GC.Collect();
+
+                    GCMemoryInfo memoryInfo1 = GC.GetGCMemoryInfo();
+
+                    for (int i = 0; i < gch.Length * 2; ++i)
+                    {
+                        byte[] arr = new byte[128];
+                        if (i % 2 == 0)
+                        {
+                            gch[i / 2] = GCHandle.Alloc(arr, GCHandleType.Pinned);
+                        }
+                    }
+
+                    // Allows to update the value returned by GC.GetGCMemoryInfo
+                    GC.Collect();
+
+                    GCMemoryInfo memoryInfo2 = GC.GetGCMemoryInfo();
+
+                    Assert.True(memoryInfo1.FragmentedBytes < memoryInfo2.FragmentedBytes);
+                    Assert.InRange(memoryInfo2.FragmentedBytes, 128 * (gch.Length - 1), long.MaxValue);
```

Comment:
The GC can still fill the holes between the pinned allocations with other allocations made during startup. If it happens, this assertion will fail.